index.js

```diff
@@ -92,26 +124,27 @@ function run() {
   cli.prepare({
     cwd: opts.cwd,
     configPath: opts.gulpfile,
-    require: opts.require,
+    preload: opts.preload,
     completion: opts.completion,
-  }, function(env) {
-    var cfgLoadOrder = ['home', 'cwd'];
-    var cfg = loadConfigFiles(env.configFiles['.gulp'], cfgLoadOrder);
-    opts = mergeConfigToCliFlags(opts, cfg);
-    env = mergeConfigToEnvFlags(env, cfg, opts);
-    env.configProps = cfg;
-
-    // Set up event listeners for logging again after configuring.
-    toConsole(log, opts);
-
-    cli.execute(env, env.nodeFlags, handleArguments);
-  });
+  }, onPrepare);
 }
 
 module.exports = run;
 
+function onPrepare(env) {
+  var cfg = mergeProjectAndUserHomeConfigs(env);
+  opts = mergeConfigToCliFlags(opts, cfg);
```

Comment:
This execution could be moved in `onExecute`. And it might be better to create a new object which is merged and filtered only necessary properties of `opts` and `cfg`.